src/packet/many.rs

```diff
@@ -83,7 +83,7 @@ impl<R: Read> Iterator for PacketParser<R> {
                             break;
                         }
                         Ok(r) => {
-                            body.extend_from_slice(&buf[..r]);
+                            body.extend_from_slice(buf.get(..r).unwrap_or(buf.as_slice()));
```

Comment:
Shouldn't it be an error if `buf` isn't long enough?

Reply:
Like @iequidoo, I also find it odd to quietly ignore an unexpectedly short `buf`.

The Reader should never report to have read more data (`r`) than fits into `buf`, I think? So throwing an Error in that case seems best to me.